features/messages/impl/src/main/kotlin/io/element/android/features/messages/impl/timeline/factories/TimelineItemsFactoryContext.kt

```diff
@@ -0,0 +1,18 @@
+/*
+ * Copyright 2024 New Vector Ltd.
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ * Please see LICENSE in the repository root for full details.
+ */
+
+package io.element.android.features.messages.impl.timeline.factories
+
+/**
+ * Some data used to configure the creation of timeline items.
+ * @param computeReadReceipts when false, read receipts will be empty.
+ * @param computeReactions when false, reactions will be empty.
+ */
+data class TimelineItemsFactoryContext(
```

Comment:
Maybe rename to `TimelineItemsFactoryParams`? Context is sort of a reserved word in Android dev...

Reply:
I find Params a bit weird here, do you have other ideas? I wasn't sure how to name it..

Reply:
I can only think of either `Params` or `Config`.

Reply:
This is a bit stupid to pass params which never changes, so I'd use @AssistedInject instead if you don't like the `lateinit`

Reply:
It's done on `init`, so it's not that bad, although it's true a refactor could easily break it. The alternative might be creating a `TimelineItemsFactoryContextBuilder` and I think none of us want to reach that level of design pattern overload 😅 .

Reply:
I don't like creating several bools for config, it should be in one wrapper IMO.
So either I pass the wrapper to the create function, but it's a bit stupid because it won't change overtime, either I also use an `AssistedFactory` 😅

Reply:
Yes, to be clean we will need an assisted factory.

Reply:
Done.